io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -436,6 +437,10 @@ instance Exception TimeoutException where
   toException   = asyncExceptionToException
   fromException = asyncExceptionFromException
 
+instance MonadEventlog (SimM s) where
+  traceEventM = traceM
+  traceMarkerM = traceM
```

Comment:
These should probably use (& export) an extra newtype wrapper so that they are distinguishable for any other `String` that gets traced via the dynamic typing scheme.

Reply:
I have added `EventlogEvent` and `EventlogMarker`. Ok?